flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/table/CorrelateITCase.scala

```diff
@@ -82,6 +82,45 @@ class CorrelateITCase extends StreamingMultipleProgramsTestBase {
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+  /**
+    * Due to CALCITE-2004, common join predicates are temporarily forbidden.
+    */
+  @Test (expected = classOf[ValidationException])
+  def testLeftOuterJoinWithPredicates(): Unit = {
```

Comment:
The validation has been checked before. We don't need an ITCase for this.

Reply:
This is the same with the batch fashion.
> This test should have been used as a normal test case for this join, i.e., we don't expect it to raise an exception. I'd suggest to keep it for later use, otherwise we may not be aware the test is incomplete in the future.

Reply:
I see, but let's rather add a JIRA issue for FLINK and link to that one (which links to the Calcite JIRA and the discussion on the Calcite dev list).